src/classes/BDI_MigrationMappingUtility.cls

```diff
@@ -428,6 +465,44 @@ public without sharing class BDI_MigrationMappingUtility {
         }
     }
 
+    /*******************************************************************************************************
+    * @description Creates and enqueues mising Migrated Custom Object Mapping records
+    * @param targetObjectMappingLabel - label of target object mapping
+    * @param targetObjectMappingRecordName - record-name of target object mapping
+    * @param dataImportObjectDeveloperNameByLabel - existent migrated object mappings
+    * @param defaultObjectMappings - existent default object mappings
+    */
+    private void createAndEnqueueMigratedCustomObject(String targetObjectMappingLabel, String targetObjectMappingRecordName, 
+                                                        Map<String, String> dataImportObjectDeveloperNameByLabel,
+                                                        Map<String, Data_Import_Object_Mapping__mdt> defaultObjectMappings) {     
+        
+        // get existent object mapping to use as reference (if exists)
+        Data_Import_Object_Mapping__mdt rdObjectMapping = defaultObjectMappings!=null ? 
+            defaultObjectMappings.get(targetObjectMappingRecordName):null;
+
+        // if reference object mapping exists 
+        if (rdObjectMapping!=null && !Test.isRunningTest()) { 
+
+            // create instance of object mapping
+            BDI_MigrationMappingUtility.DataImportObjectMapping diom = 
+                new BDI_MigrationMappingUtility.DataImportObjectMapping(rdObjectMapping, true);                
+
+            // get metadata object to be enqueued
+            Metadata.CustomMetadata objectMetadata = diom.getMetadataRecord();
+            // get unique migrated name using full name
+            String uniqueName = objectMetadata.fullName;
+            // split by '.' and get the last array node
+            uniqueName =  uniqueName.split('\\.')[uniqueName.split('\\.').size()-1] ;
```

Comment:
Just want to double check... we can't utilize the existing data import object mapping wrapper and therefore generateUniqueRecordName(), right?

Reply:
We can't 👎  The inner class attributes are defined without public attributes and without getter-setter so is not reachable. 
I didn't want to modify it given that is being used in several places.